src/pybind/mgr/volumes/module.py

```diff
@@ -417,16 +431,20 @@ def _cmd_fs_clone_cancel(self, inbuf, cmd):
         return self.vc.clone_cancel(
             vol_name=cmd['vol_name'], clone_name=cmd['clone_name'],  group_name=cmd.get('group_name', None))
 
-    def _cmd_nfs_export_create(self, inbuf, cmd):
-        #TODO Extend export creation for rgw.
-        return self.fs_export.create_export(export_type=cmd['type'], fs_name=cmd['fsname'],
-                pseudo_path=cmd['binding'], read_only=cmd.get('readonly', False),
-                path=cmd.get('path', '/'), cluster_id=cmd.get('attach'))
+    def _cmd_nfs_export_create_cephfs(self, inbuf, cmd):
+        #TODO Add similar export interface for rgw.
+        return self.fs_export.create_export(fs_name=cmd['fsname'], pseudo_path=cmd['binding'],
+                read_only=cmd.get('readonly', False), path=cmd.get('path', '/'), cluster_id=cmd.get('attach'))
 
-    def _cmd_fs_nfs_export_delete(self, inbuf, cmd):
-        return self.fs_export.delete_export(cmd['export_id'])
+    def _cmd_nfs_export_delete(self, inbuf, cmd):
+        return self.fs_export.delete_export(pseudo_path=cmd['binding'], cluster_id=cmd.get('attach'))
 
     def _cmd_nfs_cluster_create(self, inbuf, cmd):
-        #TODO add placement option
-        nfs_cluster_obj = NFSCluster(self, cmd['clusterid'])
-        return nfs_cluster_obj.create_nfs_cluster(export_type=cmd['type'], size=cmd.get('size', 1))
+        return self.nfs.create_nfs_cluster(cluster_id=cmd['clusterid'], export_type=cmd['type'],
+                                           placement=cmd.get('placement', None))
+
+    def _cmd_nfs_cluster_update(self, inbuf, cmd):
+        return self.nfs.update_nfs_cluster(cluster_id=cmd['clusterid'], placement=cmd['placement'])
+
+    def _cmd_nfs_cluster_delete(self, inbuf, cmd):
+        return self.nfs.delete_nfs_cluster(cluster_id=cmd['clusterid'])
```

Comment:
Confirm configs/recovery databases are removed. My own testing shows the recovery databases are left behind:

````
pdonnell@senta03 ~/ceph/build$ bin/rados --pool=nfs-ganesha --namespace=foo ls
2020-05-20T23:09:37.822+0000 7f54d3539cc0 -1 WARNING: all dangerous and experimental features are enabled.
2020-05-20T23:09:37.842+0000 7f54d3539cc0 -1 WARNING: all dangerous and experimental features are enabled.
2020-05-20T23:09:37.842+0000 7f54d3539cc0 -1 WARNING: all dangerous and experimental features are enabled.
grace
rec-0000000000000002:nfs.ganesha-foo.senta03
````

Reply:
Also, auth credentials for each export are left behind. Same for the recovery database credential.

Reply:
> Confirm configs/recovery databases are removed. My own testing shows the recovery databases are left behind:
> 
> ```
> pdonnell@senta03 ~/ceph/build$ bin/rados --pool=nfs-ganesha --namespace=foo ls
> 2020-05-20T23:09:37.822+0000 7f54d3539cc0 -1 WARNING: all dangerous and experimental features are enabled.
> 2020-05-20T23:09:37.842+0000 7f54d3539cc0 -1 WARNING: all dangerous and experimental features are enabled.
> 2020-05-20T23:09:37.842+0000 7f54d3539cc0 -1 WARNING: all dangerous and experimental features are enabled.
> grace
> rec-0000000000000002:nfs.ganesha-foo.senta03
> ```
Yes, I was expecting orchestrator to delete them. As I mentioned [here](https://github.com/ceph/ceph/pull/34672#discussion_r427789431). So should it be cleaned up by volumes ?
@mgfritch

Reply:
> Also, auth credentials for each export are left behind. Same for the recovery database credential.

Export auth credentials are deleted by volumes. Again I expect recovery database credentials to be deleted by orchestrator.

Reply:
You are right about export auth credentials, they are not deleted.